public/app/features/dashboard/components/GenAI/jsonDiffText.test.ts

```diff
@@ -0,0 +1,235 @@
+
+    expect(result).toEqual([]);
+  });
+
+  it('should handle deeply nested objects', () => {
+    const before = {
+      a: {
+        b: {
+          c: 'foo',
+        },
+      },
+      d: 'bar',
+    };
+
+    const after = {
+      d: 'bar',
+      a: {
+        b: {
+          c: 'foo',
+        },
+      },
+    };
+
+    const result = orderProperties(before, after);
+
+    expect(result).toEqual({
+      a: {
+        b: {
+          c: 'foo',
+        },
+      },
+      d: 'bar',
+    });
+  });
+
+  it('should handle arrays of nested objects', () => {
+    const before = [
+      { id: 1, nested: { name: 'Alice' } },
+      { id: 2, nested: { name: 'Bob' } },
+    ];
+
+    const after = [
+      { id: 2, nested: { name: 'Bob' } },
+      { id: 1, nested: { name: 'Alice' } },
+    ];
+
+    const result = orderProperties(before, after);
+
+    expect(result).toEqual([
+      { id: 1, nested: { name: 'Alice' } },
+      { id: 2, nested: { name: 'Bob' } },
+    ]);
+  });
+
+  it('should handle mixed arrays of objects and primitive values', () => {
+    const before = [{ id: 1 }, 42, [3, 2, 1]];
+
+    const after = [{ id: 1 }, [3, 2, 1], 42];
+
+    const result = orderProperties(before, after);
+
+    expect(result).toEqual([{ id: 1 }, 42, [3, 2, 1]]);
+  });
+
+  it('should handle arrays of objects with nested arrays', () => {
+    const before = [
+      { id: 1, values: [3, 2, 1] },
+      { id: 2, values: [6, 5, 4] },
+    ];
+
+    const after = [
+      { id: 2, values: [6, 5, 4] },
+      { id: 1, values: [3, 2, 1] },
+    ];
+
+    const result = orderProperties(before, after);
+
+    expect(result).toEqual([
+      { id: 1, values: [3, 2, 1] },
+      { id: 2, values: [6, 5, 4] },
+    ]);
+  });
+
+  it('should handle arrays of arrays', () => {
+    const before = [
+      [1, 2, 3],
+      [4, 5, 6],
+    ];
+
+    const after = [
+      [4, 5, 6],
+      [1, 2, 3],
+    ];
+
+    const result = orderProperties(before, after);
+
+    expect(result).toEqual([
+      [1, 2, 3],
+      [4, 5, 6],
+    ]);
+  });
+
+  it('should match reordered and modified arrays to nearest keys', () => {
+    const before = [
+      { id: '1', name: 'Alice', country: 'England' },
+      { id: '2', name: 'Bob', country: 'America' },
+      { id: '3', name: 'Charlie', country: 'Foxtrot' },
+    ];
+
+    const after: JSONArray = [{ name: 'Charlie', country: 'Foxtrot' }, { name: 'Alice' }];
+
+    const result = orderProperties(before, after);
+
+    expect(result).toEqual([{ name: 'Alice' }, { name: 'Charlie', country: 'Foxtrot' }]);
+  });
+});
+
+describe('isObject', () => {
+  it('should return true for non-array objects', () => {
+    expect(isObject({})).toBe(true);
+    expect(isObject({ foo: 'bar' })).toBe(true);
+    expect(isObject(null)).toBe(false);
+    expect(isObject([])).toBe(false);
+    expect(isObject('')).toBe(false);
+    expect(isObject(123)).toBe(false);
+    expect(isObject(true)).toBe(false);
+  });
+});
```

Comment:
I'll add some tests for `getDashboardStringDiff` to have some examples of generated diff as documentation :)